homeassistant/components/onvif/device.py

```diff
@@ -565,6 +603,50 @@ async def async_perform_ptz(
                     "Zoom": {"x": speed_val},
                 }
                 await ptz_service.RelativeMove(req)
+
+            elif move_mode == EMULATE_RELATIVE_MOVE:
+                # Using continous move, with a set time for an step
+                # Guard against unsupported operation
+                try:
+                    req = ptz_service.create_type(CONTINUOUS_MOVE)
+                    req.ProfileToken = profile.token
+                    if not profile.ptz or not profile.ptz.continuous:
+                        LOGGER.warning(
+                            "ContinuousMove not supported on device '%s'", self.name
+                        )
+                        return
```

Comment:
So does this mean the service ends? I think instead we should raise a `ServiceValidationError` with this error. This way the user gets this as feedback instead of in the logs